File: tools/namespace-generator/ginmodel/emulation_config.go
package ginmodel

type ResourceLimit struct {
	NanoCPU    string `json:"nano_cpu"`
	MemoryByte string `json:"memory_byte"`
}

type InstanceTypeConfig struct {
	Image         string            `json:"image"`
	Envs          map[string]string `json:"container_envs"`
	ResourceLimit ResourceLimit     `json:"resource_limit"`
}

type EmulationInfo struct {
	Running    bool                          `json:"running"`
	TypeConfig map[string]InstanceTypeConfig `json:"type_config"`
}

File: sub-modules/PositionProvider/satellite_emulator/model/emulation_config.py
import json
from satellite_emulator.utils.tools import object2dict

class ResourceLimit:
    def __init__(self) -> None:
        self.nano_cpu: int = 0
        self.memory_byte: int = 0

class InstanceTypeConfig:

    def __init__(self) -> None:
        self.image: str = ""
        self.envs: dict[str,str] = {}
        self.resource_limit: ResourceLimit = ResourceLimit()
        
        
class EmulationInfo:
    
    def __init__(self) -> None:
        self.running: bool = False
        self.type_config: dict[str,InstanceTypeConfig] = {}

def emulation_info_from_json(seq: str) -> EmulationInfo:
    dic = json.loads(seq)
    ret = EmulationInfo()
    ret.running = dic["running"]
    for k,v in dic["type_config"]:
        config = InstanceTypeConfig()
        config.image = v["image"]
        config.envs = v["envs"]
        config.resource_limit = ResourceLimit()
        config.resource_limit.__dict__ = v["resource_limit"]
        ret.type_config[k] = config
    return ret



def emulation_info_to_json(config: EmulationInfo) -> str:
    dic = object2dict(config)
    return json.dumps(dic)
